File: 8_Kyu_Kata/sum-mixed-array/Python/sum_mixed_array_test.py
# [object Object] - Sum Mixed Array  [ ID: 57eaeb9578748ff92a000009  (sum-mixed-array) ]
# URL: https://www.codewars.com/kata/57eaeb9578748ff92a000009
# Category: REFERENCE  |  Tags: FUNDAMENTALS | STRINGS | ARRAYS
# ******************************************************************************
import codewars_test as test
from solution import sum_mix

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(sum_mix([9, 3, '7', '3']), 22)
        test.assert_equals(sum_mix(['5', '0', 9, 3, 2, 1, '9', 6, 7]), 42)
        test.assert_equals(sum_mix(['3', 6, 6, 0, '5', 8, 5, '6', 2,'0']), 41)
        test.assert_equals(sum_mix(['1', '5', '8', 8, 9, 9, 2, '3']), 45)
        test.assert_equals(sum_mix([8, 0, 0, 8, 5, 7, 2, 3, 7, 8, 6, 7]), 61)

@test.describe("Random Tests")
def random_tests():
    
    from random import randint
    
    sol=lambda arr: (sum(int(a) for a in arr))
    
    for _ in range(40):
        arr=[randint(-100,100) if randint(0,1) else str(randint(-100,100)) for q in range(randint(1,100))]
        expected = sol(arr)
        @test.it(f"Testing for sum_mix({arr})")
        def test_case():
            test.assert_equals(sum_mix(arr), expected)

File: kata-8-kyu/sum-mixed-array/typescript/sumMixedArray.Test.ts
//+ ====================================================================================================================
//+
//+ 8 kyu - Sum Mixed Array  [ ID: 57eaeb9578748ff92a000009 ] (sum-mixed-array)
//+ URL: https://www.codewars.com/kata/57eaeb9578748ff92a000009
//+ Category: REFERENCE  |  Tags: FUNDAMENTALS | STRINGS | ARRAYS
//+
//+ ====================================================================================================================

import { assert } from "chai"
import { sumMix } from "./sumMixedArray"

describe("sumMix", () => {
  it("Example tests", () => {
    assert.equal(sumMix([9, 3, "7", "3"]), 22)
    assert.equal(sumMix(["5", "0", 9, 3, 2, 1, "9", 6, 7]), 42)
    assert.equal(sumMix(["3", 6, 6, 0, "5", 8, 5, "6", 2, "0"]), 41)
    assert.equal(sumMix(["1", "5", "8", 8, 9, 9, 2, "3"]), 45)
    assert.equal(sumMix([8, 0, 0, 8, 5, 7, 2, 3, 7, 8, 6, 7]), 61)
  })
  it("Random tests", () => {
    const randint = (a: number, b: number): number => ~~(Math.random() * (b - a + 1)) + a
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    function mysumMix(x: any[]): number {
      return x.reduce((a, b) => a + Number(b), 0)
    }
    const names = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", 1, 2, 3, 4, 5, 6, 7, 8, 9]

    for (let i = 0; i < 100; i++) {
      const x = []
      const len = randint(1, 30)
      for (let k = 0; k < len; k++) {
        x.push(names[randint(0, names.length - 1)])
      }
      assert.equal(sumMix(x), mysumMix(x), `sumMix(${x})`)
    }
  })
})
